src/main/kotlin/com/neva/gradle/fork/gui/PropertyDialogField.kt

```diff
@@ -0,0 +1,60 @@
+package com.neva.gradle.fork.gui
+
+import com.neva.gradle.fork.config.properties.Property
+import com.neva.gradle.fork.config.properties.Validator
+import java.awt.Color
+import javax.swing.*
+
+class PropertyDialogField(private val property: Property, private val propField: JComponent, private val validationMessageLabel: JLabel, private val dialog: JDialog) {
+
+  init {
+    setPropertyValue()
+  }
+
+  val name: String
+    get() = property.name
+
+  val value: String
+    get() = property.value
+
+  fun validateAndDisplayErrors(): Boolean {
+    setPropertyValue()
+    val result = property.validate()
+    if (result.hasErrors()) {
+      displayErrorState(result)
+    } else {
+      displayValidState()
+    }
+    dialog.pack()
+    return result.hasErrors()
+  }
+
+  private fun setPropertyValue() {
+    when (propField) {
```

Comment:
can we use `when` as a expression here?

```
private fun setPropertyValue() = when (propField) {
    ....
}
```

Reply:
Only if I would add an empty else block. Since when is not returning any value I would leave it as it is.